torch/serialization.py

```diff
@@ -198,6 +202,38 @@ def _cuda_deserialize(obj, location):
             return obj.cuda(device)
 
 
+def validate_hpu_device(location):
+    hpu = getattr(torch, "hpu", None)
+    assert hpu is not None, "HPU device module is not loaded"
+    device = hpu._utils._get_device_index(location)
```

Comment:
For my understanding is there a reason why we don't pass optional=True here? (just comparing with `validate_cuda_device` and curious to know if this is intentional)